src/ring_buffer.zig

```diff
@@ -20,54 +23,74 @@ pub fn RingBuffer(comptime T: type, comptime size: usize) type {
         }
 
         /// Return but do not remove the next item, if any.
-        pub fn peek(self: *Self) ?T {
+        pub fn peek(self: *Self) ?*T {
             if (self.empty()) return null;
-            return self.buffer[self.index];
+            return &self.buffer[self.index];
         }
```

Comment:
One way to eliminate the pain this change causes when using a ring buffer of pointers such as the send queue in the message bus would be to keep this function as-is and instead add a `peek_ptr()` variant that returns `?*T` which could be used when copying is not desired. The same could be done for the `iterator()` function. For the client code where we are storing struct directly in the ring buffer's array instead of pointers, using the `*_ptr()` variants would avoid copies.

What do you think about that?

Reply:
Yes, agreed. I was actually also thinking we need some way to peek/iterate a const value (via a copy that can't reach back) or pointer, in the same way that you can iterate over a slice, capturing by const value or by pointer.